cmd/ocm/create/idp/github.go

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright (c) 2020 Red Hat, Inc.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+  http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package idp
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"strings"
+
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+
+	"github.com/AlecAivazis/survey/v2"
+)
+
+func buildGithubIdp(cluster *cmv1.Cluster, idpName string) (idpBuilder cmv1.IdentityProviderBuilder, err error) {
+	clientID := args.clientID
+	clientSecret := args.clientSecret
+	organizations := args.githubOrganizations
+	teams := args.githubTeams
+	teamsOrOrgs := ""
+
+	if organizations != "" && teams != "" {
+		return idpBuilder, errors.New("GitHub IDP only allows either organizations or teams, but not both")
+	}
+
+	isInteractive := clientID == "" || clientSecret == "" || (organizations == "" && teams == "")
+
+	if isInteractive {
+		fmt.Println("To use GitHub as an identity provider, you must first register the application:")
+
+		if organizations == "" && teams == "" {
+			prompt := &survey.Input{
+				Message: "List of GitHub organizations or teams " +
+					"that will have access to this cluster",
+			}
+			err = survey.AskOne(prompt, &teamsOrOrgs)
+			if err != nil {
+				return idpBuilder, errors.New("Expected a GitHub organization or team name")
+			}
+		}
+
+		// Determine if the user entered teams or organizations
+		if strings.Contains(teamsOrOrgs, "/") {
+			teams = teamsOrOrgs
+		} else {
+			organizations = teamsOrOrgs
+		}
+
+		// Create the full URL to automatically generate the GitHub app info
+		registerURLBase := "https://github.com/settings/applications/new"
+
+		// If a single organization was listed, use that to register the application
+		if organizations != "" && !strings.Contains(organizations, ",") {
+			registerURLBase = fmt.Sprintf("https://github.com/organizations/%s/settings/applications/new", organizations)
+		} else if teams != "" && !strings.Contains(teams, ",") {
+			teamOrg := strings.Split(teams, "/")[0]
+			registerURLBase = fmt.Sprintf("https://github.com/organizations/%s/settings/applications/new", teamOrg)
+		}
+
+		registerURL, err := url.Parse(registerURLBase)
+		if err != nil {
+			return idpBuilder, fmt.Errorf("Error parsing URL: %v", err)
+		}
+
+		// Populate fields in the GitHub registration form
+		consoleURL := cluster.Console().URL()
+		oauthURL := strings.Replace(consoleURL, "console-openshift-console", "oauth-openshift", 1)
```

Comment:
This line here will not work for RHMI clusters. It also needs to change `"solution-explorer"` to `"oauth-openshift"`